core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -1023,6 +1023,12 @@ abstract class Record(private[chisel3] implicit val compileOptions: CompileOptio
           s"${this.className} does not return the same objects when calling .elements multiple times. Did you make it a def by mistake?"
         )
       }
+      if (child.binding.isDefined) {
+        throw RebindingException(
+          s"Cannot create Record ${this.className}; element ${child} of record is already a bound hardware."
```

Comment:
This PR is maybe not the right place to address this. But do we really expect users to make anything of the term `bound hardware`? Like if you ask someone at SiFive using Chisel, would they be able to tell you what it means?

Reply:
But yeah that's a good point, I think something about what it means for the thing to be bound would be helpful here

Reply:
I was commenting on the error message. I wonder if it would be better to frame this issue as `Chisel Type` vs. `Chisel Value` or something like that. `a bound hardware` seems very implementation specific. But please let these musings not hold up your PR. If we were to improve the language around this issue, we would have to update this in many places, so this would require a separate PR anyways.

Reply:
I agree with Kevin, we should have better terminology but it's a much larger issue than this PR.